### ISSUE TITLE
Cannot install dlib because of non-buildable dependency

### ISSUE THREAD

**Issue by Broxzier** [microsoft/vcpkg#4727]
When tring to get dlib using `vcpkg install dlib`, there are some dependencies, of which one is `openblas[core]:x86-windows`. This package fails to install, however, with a pretty clear message "openblas can only be built for x64 currently".

Related to, but different issue than #4673.

```
Building package openblas[core]:x86-windows...
CMake Error at ports/openblas/portfile.cmake:16 (message):
  openblas can only be built for x64 currently
Call Stack (most recent call first):
  scripts/ports.cmake:71 (include)


Error: Building package openblas:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: openblas:x86-windows
  Vcpkg version: 2018.10.20-nohash
```

**Comment by Voskrese** [comment]
`https://github.com/Microsoft/vcpkg/blob/master/ports/openblas/portfile.cmake`

forcibly ends the task \ compilation
> FATAL_ERROR

delete lines and try again

```
if(NOT VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
    message(FATAL_ERROR "openblas can only be built for x64 currently")
endif()
```

**Comment by Broxzier** [comment]
Hi Voskrese. That works for openblas, but clanpack has the exact same check:

https://github.com/Microsoft/vcpkg/blob/f1eb3aa4b95540a8fb4a8e8d341e4c08136bd756/ports/clapack/portfile.cmake#L15-L17

After removing both of these, `vcpkg install dlib` seems to finish succesfully, but is this reliable? I expect these checks to be there for a reason.

**Comment by chenzhekl** [comment]
I am facing the same issue, but with a bit different error message. Are they the same question?

```
cat /home/chenzhekl/.vcpkg/buildtrees/openblas/config-x64-linux-dbg-err.log
CMake Warning at CMakeLists.txt:59 (message):
  CMake support is experimental.  This will not produce the same Makefiles
  that OpenBLAS ships with.  Only x86 support is currently available.


CMake Error at cmake/c_check.cmake:63 (if):
  if given arguments:

    "STREQUAL" "AMD64"

  Unknown arguments specified
Call Stack (most recent call first):
  cmake/prebuild.cmake:54 (include)
  cmake/system.cmake:81 (include)
  CMakeLists.txt:62 (include)

```

**Comment by Voskrese** [comment]
vcpkg port linux **toolchain** only AMD64//x86_64//x64

Need to make changes to the assembler utility code

**Comment by LilyWangL** [comment]
Thanks for posting this issue. This issue has been fixed in PR #7888. So I am closing this issue for now.
